app/helpers/application_helper.rb

```diff
@@ -190,4 +190,35 @@ def editable_field (object, property, options={})
     }.html_safe
   end
 
+  # The default is hardcoded, here, in place, for the time being. If and
+  # when the functionality is extended, the white-list could be located in
+  # a more reasonable location in the code-base.
+  def default_param_whitelist
+      ["mode"]
+  end
+
+  def teacher_content
+    params['mode'] == 'teacher-view'
```

Comment:
I think the name should be `teacher-edition`.  
`teacher-view` could be confused with what teacher's see when previewing activities in the portal.